spec/01-unit/08-router_spec.lua

```diff
@@ -1772,94 +1772,186 @@ describe("Router", function()
 
     describe("slash handling", function()
       local checks = {
+        -- upstream url    paths           request path    expected path           strip uri  path separated
```

Comment:
there is another set of those in the integration tests, but I don't see them updated?

Reply:
It means that service path is treated as a segment, e.g. nothing is concatenated to it without separating it with /.

Reply:
@Tieske thanks for reminding about integration suite. I looked it and it doesn't even run any tests, so about a time to fix that too.

Reply:
> It means that service path is treated as a segment, e.g. nothing is concatenated to it without separating it with /.

Does this only apply for BUILDING the path, or also when MATCHING the path?

So `/hello` would match `/hello/world`, but would not match `/helloworld`, since the latter one is not a full segment.

> thanks for reminding about integration suite. I looked it and it doesn't even run any tests, so about a time to fix that too.

That is actually the more important one, since we still have to move the host and path construction from before-access to after-access. Once that is done, we no longer have the ability to run it as unit tests (of the router module), but only as integration tests.

Reply:
It is buidling only, since it is a `service` property, not a `route` property. But is does bring up the question whether `route` should get a similar option, to match only on complete segments

Reply:
it only applies to building the path.  it does not change routes in any way.

Reply:
i think that on routes you can already do it with regex.

Reply:
@Tieske the integration tests have been fixed and added. If I were to vote, I would vote `path_separated = true` to be the default. I think it is a "sane" option and a more expected one.

Reply:
No strong opinion, both have pros and cons. But since it is breaking, some consensus would be nice, so @thibaultcha @hishamhm got some cts to spare?

Reply:
No breaking changes can go into Kong 1.x.

Reply:
@hishamhm this is more complex than that.

1. Breaking change happened from `1.0` to `1.1`
2. There are more users, especially enterprise, that would like to see `1.0` behavior kept as default.
3. If we set the default to `1.0` then we will only break `1.1` users which we have less as of now, and we can say that the breaking change was a mistake (in my opinion it was).

Reply:
@bungle I would agree with you completely if this was introduced between 1.0 and 1.1, but commit 5be0335 was introduced _before_ (and included in) Kong 1.0.0 (more precisely, since Kong 1.0.0rc2).

Also, it was _known_ to be a breaking change by the time it was merged. From https://github.com/Kong/kong/pull/3780:

> The bad news is that this is a breaking change, since it changes the way the upstream path is constructed. The good news is that in the entire Kong test suite there was only 1 occurrence of a test failing because of this.

We _can_ add a flag to restore pre-1.0 behavior if we want, but we cannot change the default behavior of the Kong 1.x series.

Reply:
Ah, righ, `0.15.0` did the breakage. I for some reason though it was `1.1` that introduced the breakage. Then, well maybe we can stick with the current default even when this is going probably going to cause a lot of pain with upgrading users / customers. I know that never break rule, but then again, I also don't believe that everything is black and white even here. We have had pre `0.15` releases for years, and `1.0` only for months.